clippy_lints/src/casts/mod.rs

```diff
@@ -754,6 +755,28 @@ declare_clippy_lint! {
     "detects `as *mut _` and `as *const _` conversion"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Catches casts from a positive integer literal to some pointer type
```

Comment:
This would normally be worded as "Checks for...". Also will need to be changed with the detection code.